File: javase/src/main/java/com/fcbyk/javase/api/Math_.java
package com.fcbyk.javase.api;

// Math类所在包为java.lang包，使用的时候不需要进行导包。
// Math类包含执行基本数字运算的方法，我们可以使用Math类完成基本的数学运算
// Math类中的方法都是静态的，使用的时可直接通过类名去调用
public class Math_ {

    public static void staticAttribute(){
        System.out.println(Math.E);
        System.out.println(Math.PI);
    }

    public static void staticMethod(){
        // public static int abs(int a)         返回参数的绝对值
        System.out.println("-2的绝对值为：" + Math.abs(-2));
        System.out.println("2的绝对值为：" + Math.abs(2));

        // public static double ceil(double a)  返回大于或等于参数的最小整数
        System.out.println("大于或等于23.45的最小整数位：" + Math.ceil(23.45));
        System.out.println("大于或等于-23.45的最小整数位：" + Math.ceil(-23.45));

        // public static double floor(double a) 返回小于或等于参数的最大整数
        System.out.println("小于或等于23.45的最大整数位：" + Math.floor(23.45));
        System.out.println("小于或等于-23.45的最大整数位：" + Math.floor(-23.45));

        // public static int round(float a)     按照四舍五入返回最接近参数的int
        System.out.println("23.45四舍五入的结果为：" + Math.round(23.45));
        System.out.println("23.55四舍五入的结果为：" + Math.round(23.55));

        // public static int max(int a,int b)   返回两个int值中的较大值
        System.out.println("23和45的最大值为: " + Math.max(23, 45));

        // public static int min(int a,int b)   返回两个int值中的较小值
        System.out.println("12和34的最小值为: " + Math.min(12 , 34));

        // public static double pow (double a,double b)返回a的b次幂的值
        System.out.println("2的3次幂计算结果为: " + Math.pow(2,3));

        // public static double random()返回值为double的正值，[0.0,1.0)
        System.out.println("获取到的0-1之间的随机数为: " + Math.random());
    }
}

File: JavaScript/api/basics/Math.js
// Math 对象用于执行数学任务
// Math 对象没有构造函数
// 所有的方法和属性都是静态的

// Math 对象属性
(function(){
    console.log(Math.E);    // 返回算术常量 e，即自然对数的底数（约等于2.718）
    console.log(Math.PI);   // 返回圆周率（约等于3.14159）
})();

// Math 对象的方法
(function(){
    console.log(Math.max(5,10,4));  // 返回 x,y,z,...,n 中的最大值
    console.log(Math.min(5,10,4));  // 返回 x,y,z,...,n 中的最小值
    console.log(Math.random());     // 返回 0 ~ 1 之间的随机数
    console.log(Math.abs(-100));    // 返回 x 的绝对值
})();